lib/prism/ripper_compat.rb

```diff
@@ -238,13 +333,32 @@ def visit_number(node)
         value = yield slice[1..-1]
 
         bounds(node.location)
-        on_unary(RUBY_ENGINE == "jruby" ? :- : :-@, value)
+        on_unary(visit_unary_operator(:-@), value)
       else
         bounds(location)
         yield slice
       end
     end
 
+    if RUBY_ENGINE == "jruby" && Gem::Version.new(JRUBY_VERSION) < Gem::Version.new("9.4.6.0")
+      # JRuby before 9.4.6.0 uses :- for unary minus instead of :-@
+      def visit_unary_operator(value)
+        value == :-@ ? :- : value
+      end
+    else
+      # For most Rubies and JRuby after 9.4.6.0 this is a no-op.
+      def visit_unary_operator(value)
+        value
+      end
+    end
+
+    # Visit a binary operator node like an AndNode or OrNode
+    def visit_binary_operator(node)
+      left_val = visit(node.left)
+      right_val = visit(node.right)
+      on_binary(left_val, node.operator_loc.slice.to_sym, right_val)
```

Comment:
```suggestion
      on_binary(left_val, node.operator, right_val)
```

Reply:
Oh hey - thought I had committed this because the button was greyed out, but that was because it was outdated. Added this.